pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,188 @@
+        state.sockets = parseInt(value);
+        let divs = dividers(this.state.max / parseInt(value));
+        if (divs.indexOf(this.state.cores) === -1) {
+            state.cores = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+    onThreadsChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.threads = parseInt(value);
+        let divs = dividers(this.state.max / state.threads);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.cores = (this.state.max/(state.sockets*state.threads));
+
+        this.setState(state);
+    }
+
+    onCoresChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.cores = parseInt(value);
+        let divs = dividers(this.state.max / state.cores);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+
+    render () {
+        return (<div className="modal-body">
+            <table className="vcpu-detail-modal-table">
+                <tr>
+                    <td>
+                        <table className='form-table-ct'>
+                            <InfoRecord
+                                descr={_("vCPU Count")}
+                                tooltip={_("Specify fewer than the maximum number of virtual CPUs should be enabled.")}
+                                value={<input id="machines-vcpu-count-field" type="number" className="form-control" value={this.state.count} onChange={this.onCountSelect} />}
+                            />
+                            <InfoRecord
+                                descr={_("vCPU Maximum")}
+                                tooltip={_("Maximum number of virtual CPUs allocated for the guest OS, which must be between 1 and the maximum supported by the hypervisor.")}
+                                value={<input id="machines-vcpu-max-field" type="number" className="form-control" onChange={this.onMaxChange} value={this.state.max} />}
+                            />
+                        </table>
+                    </td>
+                    <td>
+                        <table className='form-table-ct'>
+                            <InfoRecord descr={_("Sockets")} tooltip={_("Total number of CPU sockets.")} value={
+                                <Select id='socketsSelect' value={this.state.sockets.toString()} onChange={this.onSocketChange} items={dividers(this.state.max).map((t) => t.toString())} />
+                            }/>
+                            <InfoRecord descr={_("Cores")} tooltip={_("Number of cores per socket.")} value={
```

Comment:
If you rename the label to "Cores per socket" you won't actually need this tooltip.